File: Tests/SignalR.Extras.Autofac.Test/Stubs/DynamicLifetimeHubStub.cs
using SignalR.Extras.Autofac.Test.Stubs.Dependencies;


namespace SignalR.Extras.Autofac.Test.Stubs
{

	public class DynamicLifetimeHubStub : LifetimeHub
	{

		public DynamicLifetimeHubStub(ScopedObjectStub dep1, SingletonObjectStub dep2, RequestScopedObjectStub dep3)
		{
			ScopedDependency = dep1;
			SingletonDependency = dep2;
			RequestScopedDependency = dep3;
		}

		public readonly ScopedObjectStub ScopedDependency;
		public readonly SingletonObjectStub SingletonDependency;
		public readonly RequestScopedObjectStub RequestScopedDependency;

	}

}

File: Tests/SignalR.Extras.Autofac.Test/stubs/DynamicLifetimeHubStub.fs
namespace SignalR.Extras.Autofac.Test.Stubs

open SignalR.Extras.Autofac
open SignalR.Extras.Autofac.Test.Stubs.Dependencies

type DynamicLifetimeHubStub(dep1 : ScopedObjectStub, dep2 : SingletonObjectStub) =
    inherit LifetimeHub()
    member this.ScopedDependency = dep1
    member this.SingletonDependency = dep2
